.taskcluster.yml

```diff
@@ -1,304 +1,282 @@
+            metadata:
+              name: Balrog agent docker build
+              description: Balrog agent docker build
+              owner: ${owner_email}
+              source: ${repo_url}
 
-  - provisionerId: "{{ taskcluster.docker.provisionerId }}"
-    workerType: "{{ taskcluster.docker.workerType }}"
-    scopes:
-      - secrets:get:repo:github.com/mozilla/balrog:dockerhub
-    payload:
-      maxRunTime: 3600
-      image: "taskcluster/image_builder:0.1.3"
-      features:
-        dind: true
-        taskclusterProxy: true
-      command:
-        - "/bin/bash"
-        - "-c"
-        - "git clone -b {{ event.version }} {{ event.head.repo.url }} balrog && cd balrog && scripts/push-dockerimage.sh Dockerfile {{ event.version }}"
-    extra:
-      github:
-        env: true
-        events:
-          - release
-    metadata:
-      name: Balrog Docker Image Creation (release event)
-      description: Balrog Docker Image Creation (release event)
-      owner: "{{ event.head.user.email }}"
-      source: "{{ event.head.repo.url }}"
+          - taskId: {$eval: as_slugid("backend-docker-image")}
+            created: {$fromNow: ''}
+            deadline: {$fromNow: '2 hours'}
+            provisionerId: aws-provisioner-v1
+            workerType: github-worker
+            scopes:
+              - secrets:get:repo:github.com/mozilla/balrog:dockerhub
+            routes: []
+            payload:
+              maxRunTime: 3600
+              image: "taskcluster/image_builder:0.1.3"
+              command:
+                - "/bin/bash"
+                - "-c"
+                - "git clone ${repo_url} balrog && cd balrog && git checkout ${head_sha} && scripts/push-dockerimage.sh ${docker_tags}"
+              features:
+                dind: true
+                taskclusterProxy: true
+            metadata:
+              name: Balrog backend docker build
+              description: Balrog backend docker build
+              owner: ${owner_email}
+              source: ${repo_url}
 
-  - provisionerId: "{{ taskcluster.docker.provisionerId }}"
-    workerType: "{{ taskcluster.docker.workerType }}"
-    payload:
-      maxRunTime: 1200
-      image: "node:current"
-      command:
-        - "/bin/bash"
-        - "-c"
-        - "git clone $GITHUB_HEAD_REPO_URL balrog && cd balrog && git checkout $GITHUB_HEAD_BRANCH && cd react-ui && yarn install && yarn lint && yarn test"
-    extra:
-      github:
-        env: true
-        events:
-          - pull_request.opened
-          - pull_request.synchronize
-          - pull_request.reopened
-          - push
-        branches:
-          - master
-    metadata:
-      name: Balrog React UI tests
-      description: Balrog React UI tests
-      owner: "{{ event.head.user.email }}"
-      source: "{{ event.head.repo.url }}"
+      # Deploy the UI to stage whenever we create a versioned release tag
+      - $if: 'tasks_for == "github-release" && release_tag[0] == "v"'
```

Comment:
Heads up that this means github releases where the tag mistakenly misses the prefixed "v" will silently fail to create the task

Reply:
Is there any way to make this fail not-silent? It's not ideal, I agree, but I don't want to create tasks for other types of tags (CloudOps' infrastructure will just ignore them, so it's wasteful).

Reply:
Unfortunately, the only real way to provide communication from "automation validation" -> "github release-ee" is to create a task on every release that informs the release-ee if validation fails (by logging/failing the task/leaving a comment on the release's commit). I agree that creating such a task on all releases could be wasteful,but it's worth considering the balance of "how much does it cost maintaining this task and wasting machine hours on every release" vs "how much will it cost if someone can't figure out why their release isn't triggering automation".

As I write this, I doubt the extra task is worthwhile, but it's still food for thought.